lua/lazyvim/plugins/extras/lang/tex.lua

```diff
@@ -0,0 +1,61 @@
+return {
+  {
+    "folke/which-key.nvim",
+    opts = {
+      defaults = {
+        ["<LocalLeader>l"] = { name = "+vimtex" },
+      },
+    },
+  },
+
+  -- Add BibTeX/LaTeX to treesitter
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      if type(opts.ensure_installed) == "table" then
+        vim.list_extend(opts.ensure_installed, { "bibtex", "latex" })
+      end
+      if type(opts.highlight.disable) == "table" then
+        vim.list_extend(opts.highlight.disable, { "latex" })
+      else
+        opts.highlight.disable = { "latex" }
+      end
+    end,
+  },
+
+  {
+    "lervag/vimtex",
+    version = "*",
+    ft = { "bib", "tex" },
+    keys = {
+      { "<LocalLeader>ld", "<Cmd>VimtexDocPackage<CR>", desc = "Open Documentation" },
+    },
+    config = function()
+      vim.opt.conceallevel = 2
+      vim.g.vimtex_complete_enabled = 0 -- use texlab for completion
```

Comment:
VimTeX's omni-completion is slower but more complete, as it makes full use of the Vim complete API (e.g., can search on `menu`). It's better to allow both to coexist: autocomplete via texlab, and manual omnicomplete via VimTeX.

Reply:
So your suggestion is remove line 35, am I right?

I tried to add omni-completion into nvim-cmp according to the VimTeX's document. However, the solution requires a change in `opts.formatting.format`, which is a function. Maybe I can use `require("cmp").setup.buffer()` to only config specific filetypes, but that need either `after/ftplugin` or `autocmd FileType`.

Reply:
I don't use `cmp`, sorry. But I believe there's a custom source for VimTeX?

Reply:
No, the source VimTeX using is omni. For a better result, the omni suggestions should be formatted in a function.

If leave VimTeX's completion enabled, users can still find suggestions by `<C-X><C-O>`, but I'm not sure if this is a good behaviour.

Reply:
@folke would you like to provide some advise about enable omni-completion?